### DISCUSSION TITLE
Portable Storage Database Location and Wikimedia Formatting

### DISCUSSION THREAD

**Discussion by fishercameron** [TriliumNext/Trilium#4226] [Q&A]
Hi, not familiar with coding by much but Trilium overall is interesting me and I am giving it a shot, however I have a couple use case issues. If it would be better to split this into two discussions let me know, not sure how specific these need to be but I didn't feel the need to clutter with posts.

Portable Storage Database Location:
My ideal use case is to have a portable database stored on a flashdrive/portable storage, however it doesn't look like Trilium is at all built to be able to handle this even incidentally from what I have seen, what would be the best way to approach this issue? I'd like to avoid needing to self host.

Wikimedia Formatting:
I was wondering if it were at all possible, and how I would approach, implementing anything approaching wikimedia style assets such as infoboxes. Things like two column pages, 'invisible' tables, adjustable column formatting, etc. all would be very helpful to me, but I haven't been able to find a way to do this, though it seems like it might be possible.

Thanks!

**Comment by sottey** [comment]
Hi there!

Regarding the Portable Storage Database issue:

If you had your Trilium-data directory on a portable drive (this is where the DB, config and other files are) then it should work fine. I have not tried installing everything onto a thumb drive, but since you can specify a custom data directory, this should be fine. Since the drive letter/location may change, you would want to set the data directory to a relative path (relative to the executable, I believe).

You can find out more here:
https://github.com/zadam/trilium/wiki/Data-directory

**Reply by zadam** [reply]
The key here is to set the `TRILIUM_DATA_DIR` environment variable. There's also a script to do that - e.g. `trilium-portable.bat` for windows.

**Comment by zadam** [comment]
> I was wondering if it were at all possible, and how I would approach, implementing anything approaching wikimedia style assets such as infoboxes.

Something remotely similar in purpose are Promoted attributes, especially with conjunction with Templates:

https://github.com/zadam/trilium/wiki/Promoted-attributes
https://github.com/zadam/trilium/wiki/Template

But it's nowhere close to the capabilities of MediaWiki.

If you're willing to script, then you can add your own custom widgets, but that would likely be a lot of effort.
